src/NuGet.Indexing/ResponseFormatter.cs

```diff
@@ -346,7 +346,19 @@ public static void WriteStatsResult(JsonWriter jsonWriter, NuGetIndexSearcher se
             jsonWriter.WriteStartObject();
             WriteProperty(jsonWriter, "numDocs", searcher.IndexReader.NumDocs());
```

Comment:
There doesn't need to be any complex serialization logic. Just declare the dictionary in `SearcherManager` as `IDictionary<string, Func<string>>` and directly write the transformation there. All of our diagnostics are already strings.

```
foreach (var diagnosticName in searcher.Diagnostics.Keys) {
    WriteProperty(jsonWriter, diagnosticName, searcher.Diagnostics[diagnosticName]());
}
```